### PR TITLE
Use SafeMath wherever possible

### PR DESCRIPTION
**Pull request by NicholasDotSol**
closes: #484 
Use SafeMath instead of conventional operators.

### CODE REVIEW

File: implementation/contracts/deposit/Deposit.sol [hunk 1]
```diff
@@ -275,7 +275,7 @@ contract Deposit is DepositFactoryAuthority {
     }
 
     /// @notice                 Anyone can provide a signature that was not requested to prove fraud during funding.
-    /// @dev                    ECDSA is NOT SECURE unless you verify the digest.
```
Comment (mhluongo) [inline implementation/contracts/deposit/Deposit.sol]: I kind of like this blaring reminder everywhere, is there another spot to drop it?

Comment (Shadowfiend) [reply implementation/contracts/deposit/Deposit.sol]: As-is it's not particularly actionable anyway (what does “verify” mean?); moreover, *I believe* keep-ecdsa does the verification in question, so to both consumers and developers this isn't a relevant concern when interacting with tBTC.

Worth a double-check, of course, but this is why we removed some other instances of the warning.

**Comment by Shadowfiend** [comment]
Are we worried about increasing gas costs for numbers that we already know will be safe, say because they're constants in the system or because they are governable but can have limits imposed on the governance side?

I do like safety by default, just want to make sure we've considered this.

**Comment by Shadowfiend** [comment]
Also, let's capture an issue here so we can let Diligence know, in case they've got something already in their issue pile.

File: implementation/contracts/deposit/Deposit.sol (hunk 1 above)
Comment (NicholasDotSol) [reply implementation/contracts/deposit/Deposit.sol]: yeah this was just cleaning up what was left from the original PR [comment](https://github.com/keep-network/tbtc/pull/459#discussion_r380381916)
Warning in ECDSAKeep should be good

**Comment by NicholasDotSol** [comment]
> Are we worried about increasing gas costs for numbers that we already know will be safe, say because they're constants in the system or because they are governable but can have limits imposed on the governance side?

the gas cost increase is there although really small. I still stand by using SafeMath everywhere. There is less overhead if a value changes. Also constants today might be different tomorrow.

**Pull request merged**